mongo_connector/dest_mapping.py

```diff
@@ -0,0 +1,186 @@
+                            db, col = plain_src_ns.split(".", 1)
+                            if col != "$cmd":
+                                arr = self.plain_db.get(db, [])
+                                arr.append(res.split(".")[0])
+                                self.plain_db[db] = list(set(arr))
+                            self.lock.release()
+                            return res
+                        else:
+                            self.lock.release()
+                            raise errors.InvalidConfiguration("Destination namespaces set wildcard pattern conflicts with the plain pattern.")
+            
+        if defaultval:
+            self.lock.release()
+            return defaultval
+        else:
+            self.lock.release()
+            LOG.error("Failed to find matched mapping.")
+            return None
+    
+
+    def set(self, key, value):
+        """Set the DestMapping instance to corresponding dictionary"""
+        self.lock.acquire()
+        db = key.split(".")[0]
+        if "*" in db:
+            raise errors.InvalidConfiguration("Wildcard is not allowed in db name")
+            
+        if "*" in key:
+            if key.count("*") != value.count("*"):
+                raise errors.InvalidConfiguration("The number of wildcards in the mapping should be equal")
+            self.wildcard[key] = value
+        else:
+            if value not in self.reverse_plain:
+                self.plain[key] = value
+                self.reverse_plain[value] = key
+                arr = self.plain_db.get(db, [])
+                arr.append(value.split(".")[0])
+                self.plain_db[db] = list(set(arr))
+            else:
+                raise errors.InvalidConfiguration(
+                "Destination namespaces set should not"
+                " contain any duplicates.")
+            
+        self.lock.release()
+
+    def get_key(self, plain_mapped_ns):
+        """Given a plain mapped namespace, return a source namespace if matched.
+        Only need to consider plain mappings since this is only called in rollback
+         and all the rollback namespaces in the target system should already been 
+         put in the reverse_plain dictionary.
+        """
+        return self.reverse_plain.get(plain_mapped_ns)
+    
+    def map_namespace(self, plain_src_ns):
+        """Applies the plain source namespace mapping to a "db.collection" string.
+        The input parameter ns is plain text."""
+        # if plain_src_ns matches ex_namespace_set, ignore
+        if self.match_set(plain_src_ns, self.ex_namespace_set):
+            return None       
+        # if no namespace_set, no renaming
+        if not self.namespace_set:
+            return plain_src_ns
+        # if plain_src_ns does not match namespace_set, ignore
+        elif not self.match_set(plain_src_ns, self.namespace_set):
+            return None
+        else:
+            return self.get(plain_src_ns, plain_src_ns)
```

Comment:
Can we combine these last two cases by just returning `self.get(plain_src_ns)`? If get() doesn't find plain_src_ns in self.namespace_set, it should return `None` anyway.